extensions/http-provisioner/src/main/java/org/eclipse/dataspaceconnector/transfer/provision/http/webhook/HttpProvisionerWebhookApiController.java

```diff
@@ -58,27 +57,27 @@ public void callProvisionWebhook(@PathParam("processId") String transferProcessI
                 .build();
 
         var response = ProvisionResponse.Builder.newInstance()
-                .resource(cr)
+                .resource(contentResource)
                 .secretToken(new SimpleSecretToken(request.getApiKeyJwt()))
                 .build();
-        var cmd = new AddProvisionedResourceCommand(transferProcessId, response);
 
-        transferProcessManager.enqueueCommand(cmd);
+        var command = new AddProvisionedResourceCommand(transferProcessId, response);
+
+        transferProcessManager.enqueueCommand(command);
 
     }
 
     @Override
     @POST
     @Path("/{processId}/deprovision")
     public void callDeprovisionWebhook(@PathParam("processId") String transferProcessId, @Valid DeprovisionedResource resource) {
+        if (resource == null || resource.getProvisionedResourceId() == null) {
```

Comment:
nit: the `resource == null` part could be replaced by putting the `@NotNull` annotation on the method parameter (both here and in the interface).

Reply:
As I said before, this validation thing seems to not work very well, try to remove that null check and run the `HttpProvisionerWebhookApiControllerIntegrationTest`.

Reply:
if you pick the "right" `@NotNull` annotation, and put it on the interface and the class it should work. at least it worked for me...?

Reply:
@paullatzelsperger oh, now I remember, the problem here is that the `DeprovisionedResource` object is used, and it's located in the `transfer-spi` module, so it cannot have the annotation registered. I would left this as it is and provide a dedicated DTO in another PR

Reply:
ahh, you're right. 
we can also leave the `resource == null` check in place its not really a problem :) AFAIK the `DeprovisionedResource` was originally intended to be serializable, so I'd be ok with just leaving it.